src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs

```diff
@@ -118,7 +117,68 @@ private static void OnAutoPlayChanged(DependencyObject sender, DependencyPropert
 
 		#endregion
 
-		#region MediaPlayer Property
+		#region IsFullWindow Property
+
+		public bool IsFullWindow
+		{
+			get { return (bool)GetValue(IsFullWindowProperty); }
+			set { SetValue(IsFullWindowProperty, value); }
+		}
+
+		public static DependencyProperty IsFullWindowProperty { get; } =
+			DependencyProperty.Register(
+				nameof(IsFullWindow),
+				typeof(bool),
+				typeof(MediaPlayerElement),
+				new FrameworkPropertyMetadata(false, OnIsFullWindowChanged));
+
+
+		private static void OnIsFullWindowChanged(DependencyObject sender, DependencyPropertyChangedEventArgs args)
+		{
+			sender.Maybe<MediaPlayerElement>(mpe =>
+			{
+				mpe.ToogleFullScreen((bool)args.NewValue);
+			});
+		}
+
+		private object _mainContent;
+
+		private void ToogleFullScreen(bool showFullScreen)
+		{
+			if (showFullScreen)
+			{
+				Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
+				{
+					ApplicationView.GetForCurrentView().TryEnterFullScreenMode();
+				});
+
+				_mainContent = (Windows.UI.Xaml.Window.Current.Content as Frame).Content;
```

Comment:
This is a little brittle:
1. There's no requirement for the Window.Content to be a Frame
2. If app code tries to access the visual tree while the media player is in full screen, it's going to get a surprise.
3. If there's a page navigation while the media player is full screen, the new page is effectively going to be erased when it leaves fullscreen (and the navigation stack will probably be screwed up).
etc

Did you look into doing it closer to how UWP does? Ie using FullScreenMediaRoot or whatever it's called. The implementation would probably be similar to that for Popup.

Reply:
I'll check that

Reply:
@davidjohnoliver Updated. Could you review again please?